core/server/src/main/java/alluxio/worker/file/FileSystemWorker.java

```diff
@@ -94,54 +106,73 @@ public FileSystemWorker(BlockWorker blockWorker) throws IOException {
    * Cancels a file currently being written to the under file system. The open stream will be
    * closed and the partial file will be cleaned up.
    *
+   * @param sessionId the session id of the request
    * @param tempUfsFileId the id of the file to cancel, only understood by the worker that created
    *                      the file
    * @throws FileDoesNotExistException if this worker is not writing the specified file
    * @throws IOException if an error occurs interacting with the under file system
    */
-  public void cancelUfsFile(long tempUfsFileId) throws FileDoesNotExistException, IOException {
-    mUnderFileSystemManager.cancelFile(tempUfsFileId);
+  public void cancelUfsFile(long sessionId, long tempUfsFileId)
+      throws FileDoesNotExistException, IOException {
+    mUnderFileSystemManager.cancelFile(sessionId, tempUfsFileId);
+  }
+
+  /**
+   * Cleans up after sessions, to prevent zombie sessions. This method is called periodically by
+   * {@link SessionCleaner} thread.
+   */
+  public void cleanupSessions() {
```

Comment:
I'd prefer to pass this as a callback to the cleanup thread instead of adding an interface to this class so that it can expose this method. Is the interface needed other than for exposing this method?